stdlib/Distributed/src/remotecall.jl

```diff
@@ -578,12 +600,14 @@ value to the return value of the call upon completion.
 """
 function put!(rr::Future, v)
     rr.v !== nothing && error("Future can be set only once")
+    rr.v = Some(v) # cache before put_future
```

Comment:
Maybe `put!`-`put!` race is not the main focus here, but, you'd need a CAS here to ultimately make Distributed thread-safe. Something like:

```suggestion
    _, ok = @atomicreplace rr.v nothing => Some(v)
    ok || error("Future can be set only once")
```

Reply:
Yeah I think I'll have to do that. Started hitting some issues in more advanced scenarios with multiple fetches and feels like without atomic ordering it just won't work.
Thanks!